Cogs/Helpers/websiteHelpers.py

```diff
@@ -0,0 +1,312 @@
+
+        out = utils.get(self.players, tag=text)
+        if out != None:
+            return out
+
+        for player in self.players:
+            if text == f"{player.tag} - {player.name}":
+                return player
+        return None
+
+    """
+    Attempts to create a Player.
+
+    If the server accepts the creation, the player is added to the cache and None is returned
+    Otherwise the failure message is returned.
+    """
+    async def create_player(self, name: str, tag: str) -> Union[str, None]:
+        data = {
+            "Name": name,
+            "ScreenName": tag
+        }
+        status, response = await sendPost("NewPlayer", data)
+
+        if(status == 200):
+            player = Player({"id":response["id"], "name": name, "screenName":tag})
+            print(f"Player created: {player}")
+            self.players.append(player)
+            return None
+        else:
+            return "Failed to create Player" #TODO DUPLICATE MESSAGE
+        #TODO maybe add duplicate checking
+
+class TeamCache():
+    """
+    Creates a new cache of teams from the website.
+
+    refresh is the amount of time after which the cache is purged and
+    all new data is pulled from the website
+    """
+    def __init__(self, refresh: datetime.timedelta) -> None:
+        self.teams: dict[str, list[Team]] = {}
+        self.last_updated = datetime.datetime.min
+        self.refresh_interval = refresh
+        self.lock = asyncio.Lock()
+
+    async def update_cache(self):
+        # Do thread safety
+        async with self.lock:
+            if(self.last_updated + self.refresh_interval < datetime.datetime.now() ):
+                for game in GameNames:
+                    endpoint = f"GetTeams?GameName={game}"
+
+                    status, data = await sendPost(endpoint)
+
+                    if(status == 200):
+                        self.last_updated = datetime.datetime.now() # only update if it succeeded.
+                        self.teams[game] = []
+
+                        if(len(data) > 0):
+                            print(f"Caching {len(data)} Teams")
+                            for teamData in data:
+                                self.teams[game].append(Team(teamData))
+                        else:
+                            print("Warning: No Teams retrieved into cache")
+                    else:
+                        print("Error: Failed retrieving Teams into cache")
+
+    """
+    If game is specified, removes the team from the specifed game
+    Otherwise searches through all games
+    returns True if a team was removed
+    """
+    async def remove_cached(self, team:Team, game:str = None) -> bool:
+        if(game != None):
+            try:
+                self.teams[game].remove(team)
+            except ValueError:
+                return False
+            return True
+
+        for gameName in GameNames:
+            try:
+                self.teams[gameName].remove(team)
```

Comment:
We will have a problem here when there are teams with the same name within multiple games, for instance Varsity or JV. Might need to display the options to the user with matches found and let them choose which game, or just require the game argument.

Reply:
Ah. this may just be an issue with my quick and dirty cache implementation. I think this will always have game passed in though because the website requires specifying the Game and will throw an exception if there are two teams with the same name in one game.